### PR TITLE
[Translator] Dump translation constants as tree instead of simple list

### PR DESCRIPTION
**Pull request by gepo**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | no |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

PR #14434 for 2.8 branch

### CODE REVIEW

**Comment by stof** [comment]
Please rebase your branch to get rid of this merge commit

File: src/Symfony/Component/Translation/Dumper/YamlFileDumper.php [hunk 1]
```diff
@@ -40,4 +51,12 @@ protected function getExtension()
     {
         return 'yml';
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function format(MessageCatalogue $messages, $domain)
```
Comment (aitboudad) [inline src/Symfony/Component/Translation/Dumper/YamlFileDumper.php]: I would prefer to keep `format` above the `getExtension` function :)

**Comment by aitboudad** [comment]
ping @gepo

**Comment by gepo** [comment]
sorry, I was busy, will fix it today

**Comment by gepo** [comment]
done

**Comment by aitboudad** [comment]
Thank you @gepo.

**Pull request merged**